src/palace/manager/celery/tasks/playtime_entries.py

```diff
@@ -242,30 +242,48 @@ def generate_playtime_report(
 def _fetch_distinct_eligible_data_source_names(
     session: Session,
 ) -> list[str]:
-    query = (
+    """
+    Fetches a sorted list of distinct data source names for which to produce a playback time report.
+
+    We gather data source names from two sources:
+    1. Collections with eligible protocols.
+    2. Data sources that appear in PlaytimeSummary records.
+
+    The names collected from both sources are combined, deduplicated, and then
+    returned as a sorted list.
+
+    :param session: The SQLAlchemy database session.
+    :return: A sorted list of distinct data source names.
+    """
+    eligible_collection_protocols = [
+        OPDS2API.label(),
+        OPDSForDistributorsAPI.label(),
+    ]
```

Comment:
Minor: Instead of assuming that a collections protocol is its `label`, I'd like to use the functions from the integration registry in order to get the relevant protocols to lookup in the database. That way we get deprecated protocol names, if / when we switch them in the future.

I know this was already assuming the label for the protocol name, so its fine if we want to delay this fix for later. 

Something like this (where we pass registry in from the task):

```suggestion
    # Get all protocol names (canonical + aliases) for eligible integrations
    opds2_protocols = registry.get_protocols(OPDS2API, default=False)
    opds_for_distributors_protocols = registry.get_protocols(OPDSForDistributorsAPI, default=False)
    eligible_collection_protocols = opds2_protocols + opds_for_distributors_protocols
```

Reply:
@jonathangreen I see. That's really helpful. Thanks!